### PR TITLE
Refactor and update web-embed message types 

### PR DESCRIPTION
**Pull request by lizraeli**
This PR updates the web-embed package:
- Update interface types for events
- Add and use basic type guards to determine window message event type

### CODE REVIEW

File: packages/web-embed-api/src/typeGuards.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+} from './interfaces';
+
+function isRecord(val: unknown): val is { [key: string]: unknown } {
+  return val !== null && typeof val === 'object';
+}
+
+function isString(val: unknown): val is string {
+  return typeof val === 'string';
+}
+
```
Comment (alikirec) [inline packages/web-embed-api/src/typeGuards.ts:16]: I wouldn't create this function as it just checks `string` type

**Review by alikirec — APPROVED** [review]
that looks super clean!

File: packages/web-embed-api/src/typeGuards.ts (hunk 1 above)
Comment (lizraeli) [reply packages/web-embed-api/src/typeGuards.ts:16]: Yeah I agree that how it's used now this is a bit simple to put in a function. I do have a plan for it for the future, where we can have type guards that check for interfaces. For example, having an `isShape` function:

```typescript
interface Student {
   name: string,
   age: number
}

const isStudent = isShape<Student>({ 
   name: isString,
   age: isNumber
})

function main(val: unknown) {
   if (isStudent(val)){
      // val is Student
   }
}
```
There are some existing libraries that use a similar approach (e.g. [zod](https://github.com/colinhacks/zod) and [Runtypes](https://github.com/pelotom/runtypes))

**Comment by lizraeli** [comment]
[When Formsort is loaded in an embed inside of a whitelisted domain, send answers along with events](https://trello.com/c/hdIUDa54/2054-when-formsort-is-loaded-in-an-embed-inside-of-a-whitelisted-domain-send-answers-along-with-events)

**Review by fzembow — APPROVED** [review]

**Pull request merged**